src/future_feature_flags.cpp

```diff
@@ -6,15 +6,20 @@
 
 #include "wcstringutil.h"
 
-features_t::features_t() = default;
+features_t::features_t() {
+    for (const metadata_t &md : metadata) {
```

Comment:
This seems reasonable. I guess we can also make the constructor private, to make clearer that it's never constructed beyond the `global_features` singleton.

```diff
diff --git a/src/future_feature_flags.h b/src/future_feature_flags.h
index e31b306e7..727cded47 100644
--- a/src/future_feature_flags.h
+++ b/src/future_feature_flags.h
@@ -70,4 +70,2 @@ class features_t {
 
-    features_t();
-
     features_t(const features_t &rhs) { *this = rhs; }
@@ -82,2 +80,4 @@ class features_t {
    private:
+    features_t();
+
     // Values for the flags.
```

Reply:
nevermind sorry, that won't work because we construct it in tests